wire-ios-data-model/E2EIdentity/E2EIService.swift

```diff
@@ -0,0 +1,103 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+public protocol E2EIServiceInterface {
+
+    func setupEnrollment() async throws
+    func directoryResponse(directoryData: Data) async throws -> AcmeDirectory
+    func getNewAccountRequest(previousNonce: String) async throws -> Data
+    func setAccountResponse(accountData: Data) async throws
+
+}
+
+/// This class provides an interface for CoreCrypto methods related to E2EI.
+public final class E2EIService: E2EIServiceInterface {
+
+    // MARK: - Properties
+
+    public var e2eIdentity: WireE2eIdentityProtocol?
+
+    private let coreCrypto: SafeCoreCryptoProtocol
+    private let mlsClientId: MLSClientID
+    private let userName: String
+    private let handle: String
+
+    // MARK: - Life cycle
+
+    public init(coreCrypto: SafeCoreCryptoProtocol,
+                mlsClientId: MLSClientID,
+                userName: String,
+                handle: String) {
```

Comment:
Do we really store these properties? Can't they be an argument to `setupEnrollment`?